src/main/ui/cheat-code-detector.gd

```diff
@@ -0,0 +1,36 @@
+class_name CheatCodeDetector
+extends Node
+
+signal cheat_detected
+
+const MAX_LENGTH = 32
+const CODE_KEYS := {
+	"a":"", "b":"", "c":"", "d":"", "e":"", "f":"", "g":"", "h":"", "i":"",
+	"j":"", "k":"", "l":"", "m":"", "n":"", "o":"", "p":"", "q":"", "r":"",
+	"s":"", "t":"", "u":"", "v":"", "w":"", "x":"", "y":"", "z":""
+}
+
+export (Array, String) var codes := []
+
+var _pressed_keys: Dictionary = {}
+var _string_buffer: String = ""
+
+func _input(event: InputEvent) -> void:
+	if event is InputEventKey:
+		var key_text := event.as_text().to_lower()
+		if key_text in CODE_KEYS:
+			if event.pressed:
+				if not _pressed_keys.has(key_text):
+					_key_just_pressed(key_text)
+			else:
+				_pressed_keys.erase(key_text)
```

Comment:
Maybe split this into three branches: just pressed, pressed, released

Reply:
Split into four branches. I'm confused why `pressed` needs to be a case, but the InputEvent class distinguishes between `pressed` and `just_pressed` so this doesn't appear to be an oversight.